lib/matplotlib/backends/_backend_tk.py

```diff
@@ -901,37 +901,23 @@ def set_history_buttons(self):
 
 
 class ToolTip:
-    """
-    Tooltip recipe from
-    http://www.voidspace.org.uk/python/weblog/arch_d7_2006_07_01.shtml#e387
-    """
-    @staticmethod
-    def createToolTip(widget, text):
-        toolTip = ToolTip(widget)
-        def enter(event):
-            toolTip.showtip(text)
-        def leave(event):
-            toolTip.hidetip()
-        widget.bind('<Enter>', enter)
-        widget.bind('<Leave>', leave)
-
-    def __init__(self, widget):
+    def __init__(self, widget, text):
```

Comment:
Could have a short docstring explaining its use. Instantiating a class without binding it to a variable is an unusual construct. (I'm not even sure who holds the reference to the object, but I suppose, it's through the bound methods?)

Reply:
Yes, the callbacks maintain the ToolTip alive (this was already the case previously via the enter() and leave() callbacks).
I agree that instantiating the class without binding it is rather awkward, so I decided again to go back a closure-based approach like I originally tried (but keeping the repeated window create/deletions that existed in the previous code).